x/emissions/keeper/keeper_test.go

```diff
@@ -2280,6 +2296,72 @@ func (s *KeeperTestSuite) TestGetActiveTopicsWithSmallLimitAndOffset() {
 	s.Require().NotNil(pageRes, "Next key should not be nil")
 }
 
+func (s *KeeperTestSuite) TestTopicGoesInactivateOnEpochEndBlockIfLowWeight() {
+	ctx := s.ctx
+	keeper := s.emissionsKeeper
+
+	params := types.Params{
+		MaxActiveTopicsPerBlock:         uint64(2),
+		MaxPageLimit:                    uint64(100),
+		TopicRewardAlpha:                alloraMath.MustNewDecFromString("0.5"),
+		TopicRewardStakeImportance:      alloraMath.MustNewDecFromString("1"),
+		TopicRewardFeeRevenueImportance: alloraMath.MustNewDecFromString("3"),
+	}
+	err := keeper.SetParams(ctx, params)
+	s.Require().NoError(err, "Setting parameters should not fail")
+
+	topic1 := types.Topic{Id: 1, EpochLength: 15}
+	topic2 := types.Topic{Id: 2, EpochLength: 15}
+	topic3 := types.Topic{Id: 3, EpochLength: 5}
+	topic4 := types.Topic{Id: 4, EpochLength: 5}
+
+	setTopicWeight := func(topicId uint64, revenue, stake int64) error {
+		_ = keeper.AddTopicFeeRevenue(ctx, topicId, cosmosMath.NewInt(revenue))
+		_ = keeper.SetTopicStake(ctx, topicId, cosmosMath.NewInt(stake))
+		return nil
+	}
+
+	_ = setTopicWeight(topic1.Id, 10, 10)
+	_ = keeper.SetTopic(ctx, topic1.Id, topic1)
+	_ = keeper.ActivateTopic(ctx, topic1.Id)
+
+	_ = setTopicWeight(topic2.Id, 20, 10)
+	_ = keeper.SetTopic(ctx, topic2.Id, topic2)
+	_ = keeper.ActivateTopic(ctx, topic2.Id)
+
+	// Fetch next page -- should only return topic 5
+	pagination := &types.SimpleCursorPaginationRequest{
+		Key:   nil,
+		Limit: 2,
+	}
+	activeTopics, _, err := keeper.GetIdsOfActiveTopics(ctx, pagination)
```

Comment:
Just for testing is one thing, but in general we shouldn't need `GetIdsOfActiveTopics()` anymore. Can update its existing query method on the autocli to be "get ids of active topics at block" too. Ideally we don't even need it in the tests